src/tests/e2e.ts

```diff
@@ -129,3 +141,210 @@ describe("Coinbase SDK E2E Test", () => {
     });
   }, 60000);
 });
+
+describe("Coinbase SDK Stake E2E Test", () => {
+  const requiredEnvVars = [
+    "STAKE_API_KEY_NAME",
+    "STAKE_API_PRIVATE_KEY",
+    "STAKE_ADDRESS_ID_1",
+    "STAKE_ADDRESS_ID_2",
+    "STAKE_VALIDATOR_ADDRESS_1",
+  ];
+
+  beforeAll(() => {
+    dotenv.config();
+
+    requiredEnvVars.forEach(envVar => {
+      if (!process.env[envVar]) {
+        throw new Error(`Required environment variable ${envVar} is not set`);
+      }
+    });
+
+    Coinbase.configure({
+      apiKeyName: process.env.STAKE_API_KEY_NAME as string,
+      privateKey: process.env.STAKE_API_PRIVATE_KEY as string,
+    });
+  });
+
+  it("should be able to access environment variables", () => {
+    requiredEnvVars.forEach(envVar => {
+      expect(process.env[envVar]).toBeDefined();
+    });
+  });
+
+  describe("Stake: Reward Tests", () => {
+    it("should list shared eth staking rewards via StakingReward.list", async () => {
+      const networkId = Coinbase.networks.EthereumMainnet;
+      const assetId = Coinbase.assets.Eth;
+      const addressIds = [process.env.STAKE_ADDRESS_ID_1 as string];
+      const startTime = new Date(2024, 4, 1).toISOString();
+      const endTime = new Date(2024, 4, 20).toISOString();
+      const rewards = await StakingReward.list(networkId, assetId, addressIds, startTime, endTime);
+
+      expect(rewards).toBeDefined();
+      // TODO(rohit): Fix code. This should ideally return 20.
+      expect(rewards.length).toEqual(19);
```

Comment:
Is this a backend change that needs to be made? 

One thing to note is that this E2E test will run against many different backend versions as we aim for backwards compatibility.

So we typically run our E2E tests against the last N SDK versions (to simulate developers using older SDK versions). The N depends on the particular domain / product guaranteed stability.

So if this will change / be fixed in the context of a backend change, we might want to make this test less specific?

Reply:
I could change it to be >=19 and <=20 for now

Reply:
ok the API might be doing the right thing, but IMO it's confusing for end users.

Anyways, have updated the test to update the window from beginning time of start date to end time of end date. This does yield correct results.